staging/vhost-device-gpu/src/virt_gpu.rs

```diff
@@ -48,7 +47,7 @@ fn sglist_to_rutabaga_iovecs(
     for &(addr, len) in vecs {
         let slice = mem.get_slice(addr, len).unwrap();
         rutabaga_iovecs.push(RutabagaIovec {
-            base: slice.as_ptr() as *mut c_void,
+            base: slice.ptr_guard_mut().as_ptr() as *mut c_void,
```

Comment:
Not sure about this, I have to investigate this a bit more.

Reply:
in the deprecated attribute they suggest 
```
    #[deprecated(
        since = "0.12.1",
        note = "Use `.ptr_guard()` or `.ptr_guard_mut()` instead"
    )]
```
`slice.ptr_guard_mut().as_ptr() as *mut c_void` still provides a raw mutable pointer to the slice, but it tracks the lifetime of the slice and prevents further access to the slice when the PtrGuardMut object is dropped.

Reply:
and we have the xen feature enabled in our cargo.toml no? for curiosity I found this https://github.com/rust-vmm/vm-memory/commit/3adf67b4aa0da7b4f81a098b67e5681e7277119b you can also reach out to the maintainer.

Reply:
I am pretty sure we don't have `xen` enabled.  With `xen`, I think dropping the `PtrGuardMut` invalidates the ptr returned from the call to the `PtrGuardMut::as_ptr`. This would cause UB when rutabaga uses the ptr.

But I don't think we need to worry about this, since we aren't using `xen`. (I have a vague idea that  `xen` is a type-1 hypervisor, but I have no clue what the vhost crate has to do with it).

You can leave it the way it is.